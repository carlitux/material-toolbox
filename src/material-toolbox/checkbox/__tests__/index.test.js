// @flow
import React from 'react';
import { shallow, mount } from 'enzyme';

import Checkbox from '../index';

describe('component::BoundedRipple', () => {
  it('Should match snapshoot cssOnly', () => {
    const component = shallow(<Checkbox cssOnly />);
    expect(component).toMatchSnapshot();
  });

  it('Should match snapshoot', () => {
    const component = shallow(<Checkbox />);
    expect(component).toMatchSnapshot();
  });

  it('Should match snapshoot indeterminate', () => {
    const component = mount(<Checkbox indeterminate />);
    expect(component).toMatchSnapshot();
  });
});
