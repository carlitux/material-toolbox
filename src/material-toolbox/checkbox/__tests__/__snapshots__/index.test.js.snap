// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::BoundedRipple Should match snapshoot 1`] = `
<div
  className="mdc-checkbox mdc-checkbox--upgraded"
  style={Object {}}
>
  <input
    className="mdc-checkbox__native-control"
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="mdc-checkbox__background"
  >
    <svg
      className="mdc-checkbox__checkmark"
      viewBox="0 0 24 24"
    >
      <path
        className="mdc-checkbox__checkmark-path"
        d="M1.73,12.91 8.1,19.28 22.79,4.59"
        fill="none"
        stroke="white"
      />
    </svg>
    <div
      className="mdc-checkbox__mixedmark"
    />
  </div>
</div>
`;

exports[`component::BoundedRipple Should match snapshoot cssOnly 1`] = `
<div
  className="mdc-checkbox"
  style={Object {}}
>
  <input
    className="mdc-checkbox__native-control"
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="mdc-checkbox__background"
  >
    <svg
      className="mdc-checkbox__checkmark"
      viewBox="0 0 24 24"
    >
      <path
        className="mdc-checkbox__checkmark-path"
        d="M1.73,12.91 8.1,19.28 22.79,4.59"
        fill="none"
        stroke="white"
      />
    </svg>
    <div
      className="mdc-checkbox__mixedmark"
    />
  </div>
</div>
`;

exports[`component::BoundedRipple Should match snapshoot indeterminate 1`] = `
<Checkbox
  cssOnly={false}
  disabled={false}
  indeterminate={true}
>
  <div
    className="mdc-checkbox mdc-checkbox--upgraded"
    style={Object {}}
  >
    <input
      className="mdc-checkbox__native-control"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        viewBox="0 0 24 24"
      >
        <path
          className="mdc-checkbox__checkmark-path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
          stroke="white"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
  </div>
</Checkbox>
`;
